src/frontend/js/page-timing.js

```diff
@@ -54,13 +55,33 @@ PageTiming.trackPage = function(eventId) {
         PageTiming.pageloadTime = performance.timing.responseEnd - performance.timing.navigationStart;
     }
 
-    // Set up so that end time will be recorded
-    $(window).on('pagehide', function() { PageTiming.reportEndTime(); });
+    // Set up so that end time will be recorded when user turns pages.
+    // https://developer.mozilla.org/en-US/docs/Web/API/Window/pagehide_event
+    $(window).on('pagehide', PageTiming.windowEventListener);
+
+    // Record end time when document is hidden.
+    // https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon
+    // https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilitychange_event.
+    document.addEventListener('visibilityChange', PageTiming.docEventListener);
```

Comment:
I am concerned about this - visibilityChange triggers on things like just switching to a different tab in the browser, which does not mark the end of a pageview.

Reply:
My experience was that `visibilityChange` events only happened when you closed the "Clusive" tab or logged out.  Switching to a different tab caused `pagehide` events only, which the previous version was already tracking. 

However, let me double-check that.

Reply:
> However, let me double-check that.

Here's what I see in the log:
- `pageHide` events when switching between Clusive tabs (e.g., from "Dashboard" to "Library"),
- `PageTiming.handleFocusChange()` when switching among browser tabs (not `pageHide` as I thought above, but the machinery for tracking inactive time),
- `visibilityChange` events with`document.visibilityState = 'hidden'` when closing the "Clusive" browser tab, or on logout.  Actually, I can't see that in the log when closing the tab since the log window closes with it.  I'm making the assumption that because the `PageEvent` is updated in the database that the event occurred.

With respect to `visibilityChange`, what I see in the logs contradicts the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilitychange_event#usage_notes) stating that the event occurs when "...a user navigates to a new page, switches tabs, ...", as you noted in your comment.

On the other hand, `visibilityChange` is also [described elsewhere](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/sendBeacon#sending_analytics_at_the_end_of_a_session) as the most reliable way to detect the end of session for sending diagnostics, which is what we want to use it for.

I'll see if there is a way to distinguish between moving among browser tabs vs. closing the tab entirely.

Reply:
BTW there is a 'persist logs' option in the settings for the javascript console.

Reply:
Oh, but persist logs does not help if you close the tab completely. Right.

Reply:
OK, in my experimentation a 'visibilityChange' event, as written on line 65, never fires.  If I listen for 'visibilitychange', all lowercase, it fires when you switch tabs, completely hide the web browser, etc.  Let's stick with pagehide for now; although I do see the warnings that say it's unreliable on mobile.